### DISCUSSION TITLE
Enable/Disable integration

### DISCUSSION THREAD

**Discussion by alexdelprete** [frenck/spook#77] [Q&A] [answered]
Hi @frenck,

a couple of questions on the `enable/disable integration` service:

1. I used `homeassistant.reload_config_entry` for 1 year to make sure my custom component is enabled at sunrise when it goes disabled during the night polling my PV Inverter. Is the `homeassistant.enable_config_entry` offered by Spook the same or does it provide more benefits?
2. In order to enable/disable the integration, do I use any of the entity_id associated with the custom component/integration, like I do with `reload_config_entry` or is there a better way to point to the target integration?

Thanks for the help.

**Comment by frenck** [comment]
> I used homeassistant.reload_config_entry for 1 year to make sure my custom component is enabled at sunrise when it goes disabled during the night polling my PV Inverter. Is the homeassistant.enable_config_entry offered by Spook the same or does it provide more benefits?

I'm not following that question. Reloading, isn't the same thing as enabling/disabling in any way. 

> In order to enable/disable the integration, do I use any of the entity_id associated with the custom component/integration, like I do with reload_config_entry or is there a better way to point to the target integration?

It needs a config entry ID.

**Reply by alexdelprete** [reply]
> Reloading, isn't the same thing as enabling/disabling in any way.

I know, but I used it for a year to re-enable the integration when it gets disabled, so I was wondering if it's not recommended to use it to enable.

> It needs a config entry ID.

I just tried it, updating the existing automation: it allows you to pick up the integration in the listbox, that is perfect, so no need to find the config entry ID. Very easy and spooky.

Thanks Frenck, let's see tomorrow morning if it does its job as expected. :)

**Comment by frenck** [comment]
> but I used it for a year to re-enable the integration when it gets disabled

I think you are misunderstanding what a disabled config entry is. Calling a reload on a disable config entry will not enable it. Calling reload on a disable config entry is a no-op.

../Frenck

**Reply by alexdelprete** [reply]
I can tell you that it re-enables the integration, and it did for 1 year. Maybe as an unintended consequence, but it actually works. If you want I can prepare a quick video to show you.

I found this reload config thing on the forum 1y ago, can't remember the user, but he was right, and it has served well for the scope until I now changed the automation using Spook.

I also noticed Spook offers a disable device service: the integration I disable creates only 1 device (the solar inverter), is it better to disable the device or the integration itself?

**Reply by frenck** [reply]
> Maybe as an unintended consequence, but it actually works.

Sounds like a bug, the code doesn't tell me that story at least.

>  is it better to disable the device or the integration itself?

Depends on the integration and use case. There is no right/wrong/better/worse.

**Reply by alexdelprete** [reply]
> Depends on the integration and use case.

The problem is common with integrations for solar inverters: when you don't use accumulators, at sunset, the inverter shuts down completely, so the integration continues polling the inverter producing a lot of warnings in the log. In order to avoid this I would like to enable/disable the integration (or device) via automation.

**Reply by frenck** [reply, marked as answer]
> so the integration continues polling the inverter producing a lot of warnings in the log.

An integration should not do such a thing. As per developer documentation, it should ideally report that once on losing connection and once after restoring the connection. Integration should not spam the log files.

**Reply by alexdelprete** [reply]
Makes sense. Thanks for the feedbacks.
